core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala

```diff
@@ -148,16 +149,15 @@ private[spark] class TorrentBroadcast[T: ClassTag](obj: T, id: Long)
         val pieceId = BroadcastBlockId(id, "piece" + i)
         val bytes = new ChunkedByteBuffer(block.duplicate())
         if (!blockManager.putBytes(pieceId, bytes, MEMORY_AND_DISK_SER, tellMaster = true)) {
-          throw new SparkException(s"Failed to store $pieceId of $broadcastId " +
-            s"in local BlockManager")
+          throw SparkCoreErrors.storeBlockError(pieceId)
         }
       }
       blocks.length
     } catch {
       case t: Throwable =>
         logError(s"Store broadcast $broadcastId fail, remove all pieces of the broadcast")
         blockManager.removeBroadcast(id, tellMaster = true)
-        throw t
+        throw SparkCoreErrors.storeBlockError(broadcastId, t)
```

Comment:
Two issues with this:
a) We are changing the exception being thrown (note: we are catching `Throwable`).
b) Message for exception thrown in L151 is getting changed in L160.

Reply:
Will change back to throw the caught Throwable here.